testrtc/p2p-1-1.js

```diff
@@ -1,25 +1,77 @@
 const agentName = process.env.RTC_AGENT_NAME;
 const agentType = Number(process.env.RTC_IN_SESSION_ID);
-const agentSessionName = process.env.RTC_SESSION_NAME;
-const agentNumber = process.env.RTC_AGENT_NUM;
 const sec = 1000;
-const baseURL = "[YOUR-DEPLOYMENT-URL]";
+const baseURL = process.env.RTC_SERVICE_URL;
 
+// Resolution to get from the camera. 
+const videoWidthMin = 320;
+const videoHeightMin = 180;
+const videoWidthIdeal = 1280;
+const videoHeightIdeal = 720;
+const videoWidthMax = 3840;
+const videoHeightMax = 2160;
+const videoWidthExact = 1920;
+const videoHeightExact = 1080;
+
+// In SFU mode, this limits above layers. In P2P mode, this is the bandwidth cap
+const setBandwidth = "NO_CAP";
+// When to switch to SFU mode
+const sfuSwitchover = 4;
+
+// VP8 is Dailys default codec. We also support h264 in P2P and SFU mode and VP9 in P2P mode.
+const codec = "VP8"
+
+const roomProps = {
+    sfu_switchover: sfuSwitchover,
+};
+
+const setBandwidthProps = {
+    kbs: setBandwidth,
+};
+
+const callConfigProps = {
+    dailyConfig: {
+        userMediaVideoConstraints: {
+            width: {
+                "min": videoWidthMin,
+                "ideal": videoWidthIdeal,
+                "max": videoWidthMax
+            },
+            height: {
+                "min": videoHeightMin,
+                "ideal": videoHeightIdeal,
+                "max": videoHeightMax
+            }
+        },
+        userMediaAudioConstraints: {
+            autoGainControl: true,
+            echoCancellation: true,
+            noiseSuppression: true
+        }
+    }
+}
+
+
+const roomPropsData = JSON.stringify(roomProps);
+const callConfigData = JSON.stringify(callConfigProps);
+const setBandwidthData = JSON.stringify(setBandwidthProps);
+
+// Creates a room and joins it
 if (agentType === 1) {
     createAndJoinRoom();
     return;
 }
 
+// Waits for agent 1 to provide a room url
 client
     .rtcWaitForSessionValue('roomURL', function(url) {
-        console.log("joining room", agentType, agentName, url)
         joinRoom(agentName, url);
     }, 30 * sec);
 
 
 function createAndJoinRoom(agentName) {
     setExpectations();
-    const url = baseURL;
+    const url = `${baseURL}?roomParams=${roomPropsData}&callConfig=${callConfigData}&setBandwidth=${setBandwidthData}&codec=${codec}`;
```

Comment:
Maybe it makes sense to use [URLSearchParms](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) here instead of constructing a string? That way we don't have to worry about making a typo and forgetting to add a `&`. I'm fine with keeping it the way it is though

Reply:
Good idea, will look into that tomorrow